assets_src/lua/constants.lua

```diff
@@ -7,6 +7,24 @@ Constants.populationPerVillage = 4
 Constants.populationPerHQ = 8
 Constants.HPotValue = 25
 Constants.GPotValue = 25
+Constants.salvageValueReturn = 0.5
+Constants.salvageables = {
+                            "hq_foundation",
+                            "port_foundation",
+                            "barracks_foundation",
+                            "tower_foundation",
+                            "city_foundation",
+                            "water_city_foundation",
+                                                
+                            "hq",
+                            "port",
+                            "barracks",
+                            "tower",
+                            "city",
+                            "water_city",
+                            "gate",
```

Comment:
you can remove gate. Also, instead of adding them here, I'd recommend adding "Salvageable" as a tag in unitClass. similar to "foundation". Then you can get unitclass in the verb and see if "Salvageable" is in the unitClass's tag.  This makes it self contained, and if we need to add new structures in future, we don't have to modify this file. Return value can be left here.